lib/asciidoctor-bibtex/js/citeproc.rb

```diff
@@ -0,0 +1,60 @@
+module AsciidoctorBibtex
+  `const {Cite, plugins} = require('@citation-js/core')`
+  `require('@citation-js/plugin-csl')`
+  `const { styles } = require('csl-js')`
+  `const path = require('path')`
+
+  module CiteProc
+    class Processor
+      attr_reader :style, :format, :locale
+
+      def initialize(options) 
+        @style = options[:style]
+        @format = options[:format]
+        @locale = options[:locale]
+
+        styleFilePath = "../vendor/styles/#{@style}.csl"
+        styleFilePath = `path.resolve(__dirname, #{styleFilePath})`
```

Comment:
@Mogztter I needed to add this, so it works regardless of the structure. I'm sure this can be written in ruby, but I'm not quite sure how. I tried a few things like `styleFilePath == File.expand_path(styleFilePath, __FILE__)`.

Reply:
What `File.expand_path(styleFilePath, __FILE__)` gives you?

Reply:
I think it might be a bug in opal. I tried following code snippet: 

```
puts __FILE__
puts File.expand_path('../vendor/styles',  __FILE__)
```

With ruby `__FILE__` just gives the file name and `expand_path` the full absolute path. 

Wit opal (our version & newest) you get the same behaviour for `__FILE__`, but `expand_path` just gives `./vendor/styles`. 

So it seems `expand_path` or `__FILE__` is not correctly implemented.

Reply:
It would be better to use File.absolute_path. File.expand_path has some magic behavior that isn't needed here.